### ISSUE TITLE
Point collection catalog yields warnings

### ISSUE THREAD

**Issue by dopplershift** [Unidata/siphon#28]
The following code:
```python
from siphon.catalog import TDSCatalog
cat = TDSCatalog('http://thredds.ucar.edu/thredds/catalog/nws/metar/ncdecoded/catalog.xml?'
                 'dataset=nws/metar/ncdecoded/Metar_Station_Data_fc.cdmr')
```
yields:
```
WARNING:root:controlledVocabulary must have an attribute: vocabulary
WARNING:root:controlledVocabulary must have an attribute: vocabulary
```
I'm not sure if we're wrong or if thredds is wrong, but since we control it all, we really should make these warning go away somehow. (i.e. fix TDS or fix Siphon).

**Comment by JohnLCaron** [comment]
where does this message come from? an XML validator i assume. where does it
get the catalog schema?

**Comment by JohnLCaron** [comment]
<xsd:complexType name="controlledVocabulary">
  <xsd:simpleContent>
    <xsd:extension base="xsd:string">
      <xsd:attribute name="vocabulary" type="xsd:string"/>
    </xsd:extension>
  </xsd:simpleContent>
</xsd:complexType>


Optional and Required Attributes

Attributes are optional by default. To specify that the attribute is
required, use the "use" attribute:
<xs:attribute name="lang" type="xs:string" use="required"/>

**Comment by lesserwhirls** [comment]
I got it wrong in siphon. I'll get it fixed.

**Comment by lesserwhirls** [comment]
@JohnLCaron - If attributes are optional by default, what is the difference between the optional attributes that minOccurs=0 and some those that do not?

For example:

```
  <xsd:complexType name="spatialRange">
   <xsd:sequence>
     <xsd:element name="start" type="xsd:double"  />
     <xsd:element name="size" type="xsd:double" />
     <xsd:element name="resolution" type="xsd:double" minOccurs="0" />
     <xsd:element name="units" type="xsd:string" minOccurs="0" />
   </xsd:sequence>
  </xsd:complexType>
```

What is the difference between, say, start and stop, and resolution and units? Does that indicate that if there is a start and stop, there may be a resolution and units (but start and stop must exist)?

**Comment by JohnLCaron** [comment]
attributes can only be required or not.

elements (as above) can have 0 to many, default i think is min=1, max=1

**Comment by lesserwhirls** [comment]
Ah, yes. My bad for mixing up attrs and elements.

Thanks!

**Comment by dopplershift** [comment]
Fixed with 7f070120360756308e5520d6d60f4cbb313c3dd0.
